### ISSUE TITLE
Using random with specific country codes repeatedly fails

### ISSUE THREAD

**Issue by noamshalev** [mdomke/schwifty#206]
Running on Mac:
```
>>> from schwifty import IBAN
>>> IBAN.random(country_code='MU')

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/homebrew/lib/python3.11/site-packages/schwifty/iban.py", line 192, in random
    return cls.from_bban(bban.country_code, bban)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/schwifty/iban.py", line 104, in from_bban
    return cls(
           ^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/schwifty/iban.py", line 81, in __init__
    self.validate(validate_bban)
  File "/opt/homebrew/lib/python3.11/site-packages/schwifty/iban.py", line 220, in validate
    self._validate_format()
  File "/opt/homebrew/lib/python3.11/site-packages/schwifty/iban.py", line 236, in _validate_format
    raise exceptions.InvalidStructure(
schwifty.exceptions.InvalidStructure: Invalid BBAN structure: 'KUOD7879880431469331000000' doesn't match '4!a2!n2!n12!n3!n3!a'
```

It also fails with other country codes such as 'KM'.

**Comment by mdomke** [comment]
Hey @noamshalev, 
thank you for raising this issue. The thing with Mauritius (`MU`) is that its BBAN structure `MUkk bbbb bbss cccc cccc cccc 000m mm` has the suffix `000mmm`, which is supposed to be the currency code (e.g. `MUR`) preceded by three zeros. This is currently not correctly accounted for in the code to generate random BBANs. I will have a look at this.

**Comment by noam-multyx** [comment]
Thanks @mdomke for looking into it.
Leaving it for you do decide, but here is my opinion:

I assume that people have the following expectations from the random generator:
 - Output a number that matches the requirements  (e.g., match the requested country)
 - The output number passes the basic sanity checks (in other words, passes the `IBAN.validate()` test)

For example: for MU, the following code works
```
>>> from schwifty import IBAN
>>> myiban = IBAN('MU43BOMM0101123456789101000MUR')
>>> myiban.validate()
True
```

Hence, for cases where we have any challenge with the structure / layout, I wouldn't throw an exception, rather generate a number the holds the **minimal** requirements that we can provide and passes the above-mentioned sanity check.

**Comment by noamshalev** [comment]
Thanks @mdomke for looking into it.
Leaving it for you do decide, but here is my opinion:

I assume that people have the following expectations from the random generator:
 - Output a number that matches the requirements  (e.g., match the requested country)
 - The output number passes the basic sanity checks (in other words, passes the `IBAN.validate()` test)

For example: for MU, the following code works
```
>>> from schwifty import IBAN
>>> myiban = IBAN('MU43BOMM0101123456789101000MUR')
>>> myiban.validate()
True
```

Hence, for cases where we have any challenge with the structure / layout, I wouldn't throw an exception, rather generate a number the holds the **minimal** requirements that we can provide and passes the above-mentioned sanity check.

**Comment by mdomke** [comment]
This issue should be fixed with release `2024.05.4`. Thanks again for reporting.
